apps/bandwagon/views.py

```diff
@@ -370,12 +370,15 @@ def ajax_list(request):
         return http.HttpResponseBadRequest()
 
     # Get collections associated with this user
-    collections = Collection.objects.publishable_by(request.amo_user)
-
-    for collection in collections:
-        # See if the collections contains the addon
-        if addon_id in collection.addons.values_list('id', flat=True):
-            collection.has_addon = True
+    has_addon = """
+        select 1 from addons_collections as ac
+            where ac.addon_id = %s and ac.collection_id = collections.id
```

Comment:
doesn't really make any difference since we'll catch an invalid value on top, but we can do that here anyway if you'd like.